ddtrace/internal/processor/stats.py

```diff
@@ -0,0 +1,239 @@
+from collections import defaultdict
+import threading
+import typing
+
+import ddtrace
+from ddtrace import config
+from ddtrace.vendor.ddsketch.ddsketch import LogCollapsingLowestDenseDDSketch
+from ddtrace.vendor.ddsketch.pb.proto import DDSketchProto
+
+from . import SpanProcessor
+from ...constants import SPAN_MEASURED_KEY
+from .._encoding import packb
+from ..agent import get_connection
+from ..compat import get_connection_response
+from ..compat import time_ns
+from ..hostname import get_hostname
+from ..logger import get_logger
+from ..periodic import PeriodicService
+from ..writer import _human_size
+
+
+if typing.TYPE_CHECKING:
+    from typing import DefaultDict
+    from typing import List
+    from typing import Optional
+
+    from ddtrace import Span
+
+    from ..agent import ConnectionType
+
+
+log = get_logger(__name__)
+
+
+def _is_top_level(span):
+    # type: (Span) -> bool
+    """Return whether the span is a "top level" span."""
+    return (span._local_root is span) or (
+        span._parent is not None and span._parent.service != span.service and span.service is not None
+    )
+
+
+def _is_measured(span):
+    # type: (Span) -> bool
+    """Return whether the span is flagged to be measured or not."""
+    return SPAN_MEASURED_KEY in span.metrics
```

Comment:
what if it is `_dd.measured: 0` ?

Reply:
Hmm not sure if we ever do this but worth being explicit :+1: